lib/stdlib/src/shell.erl

```diff
@@ -1092,21 +1121,68 @@ local_func(f, [{var,_,Name}], Bs, _Shell, _RT, _FT, _Lf, _Ef) ->
     {value,ok,erl_eval:del_binding(Name, Bs)};
 local_func(f, [_Other], _Bs, _Shell, _RT, _FT, _Lf, _Ef) ->
     erlang:raise(error, function_clause, [{shell,f,1}]);
-local_func(fl, [], Bs, _Shell, _RT, FT, _Lf, _Ef) ->
-    {value, ets:tab2list(FT), Bs};
-local_func(fd, [{atom,_,FunName}, FunExpr], Bs, _Shell, _RT, FT, _Lf, _Ef) ->
+%% Output local functions (with specs)
+local_func(lf, [], Bs, _Shell, _RT, FT, _Lf, _Ef) ->
+    Output = local_functions_and_specs(FT),
+    io:requests([{put_chars, unicode, Output++"\n"}, nl]),
+    {value, ok, Bs};
+%% Output local types
+local_func(lt, [], Bs, _Shell, _RT, FT, _Lf, _Ef) ->
+    Output = local_types(FT),
+    io:requests([{put_chars, unicode, Output}, nl]),
+    {value, ok, Bs};
+%% Output local records
+local_func(lr, [], Bs, _Shell, _RT, FT, _Lf, _Ef) ->
+    Output = local_records(FT),
+    io:requests([{put_chars, unicode, Output}, nl]),
+    {value, ok, Bs};
+
+%% Undocumented function that outputs contents of FT to a module file
+%% compiles it and loads it, its still in experimentation phase
+%% In theory, you may want to be able to load a module in to local table
+%% edit them, and then save it back to the file system.
+%% You may also want to be able to save a test module.
+local_func(save_module, [{string,_,PathToFile}], Bs, _Shell, _RT, FT, _Lf, _Ef) ->
+    [_Path, FileName] = string:split("/"++PathToFile, "/", trailing),
+    [Module, _] = string:split(FileName, ".", leading),
+    Output = (
+        "-module("++Module++").\n\n" ++
+        "-export(["++[atom_to_list(F)++"/"++integer_to_list(A)||{F,A}<-local_defined_functions(FT)]++"]).\n\n"++
```

Comment:
```suggestion
       Exports = [atom_to_list(F)++"/"++integer_to_list(A)||{F,A}<-local_defined_functions(FT)],
        "-export(["++lists:join(",",Exports)++"]).\n\n"++
```

Need to add `,` inbetween exports. Also should probably be a test for multiple functions when saving module to make sure it works.